dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/sync/EventSynchronization.java

```diff
@@ -187,6 +239,7 @@ protected void synchronizePage(int page, EventSynchronisationContext context) {
           events.getEvents().stream().map(Event::getEvent).collect(Collectors.toList());
       log.info("The lastSynchronized flag of these Events will be updated: " + eventsUIDs);
       eventService.updateEventsSyncTimestamp(eventsUIDs, context.getStartTime());
+      return;
```

Comment:
this fixes a separate bug introduced in [this PR](https://github.com/dhis2/dhis2-core/pull/10402/files#diff-d4fc2728be8a0844b9c927cd5a58e1af6db9ac0bcae6edc35e7711d83956017a), whereby a `MetadataSyncServiceException` would always be thrown in `EventSynchronization#synchronizePage`.
A `return` is needed to avoid always throwing the exception.